### ISSUE TITLE
ATOMIC_VAR_INIT is deprecated in Xcode 16

### ISSUE THREAD

**Issue by FigBug** [kfrlib/kfr#233]
ATOMIC_VAR_INIT is deprecated in Xcode 16

Please replace:

```
    std::atomic_uintptr_t allocation_count   = ATOMIC_VAR_INIT(0);
    std::atomic_uintptr_t allocation_size    = ATOMIC_VAR_INIT(0);
    std::atomic_uintptr_t deallocation_count = ATOMIC_VAR_INIT(0);
    std::atomic_uintptr_t deallocation_size  = ATOMIC_VAR_INIT(0);
```
with
```
    std::atomic_uintptr_t allocation_count   = {0};
    std::atomic_uintptr_t allocation_size    = {0};
    std::atomic_uintptr_t deallocation_count = {0};
    std::atomic_uintptr_t deallocation_size  = {0};
```

**Comment by dancazarin** [comment]
Fixed in the dev branch. Once testing is finished, it will be merged into the main branch
